### PR TITLE
Add Cargo feature to disable linker script

### PR DESCRIPTION
**Pull request by hannobraun**
This is useful if the user wants to supply their own linker script.

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -25,24 +25,26 @@ fn main() {
...
+            unreachable!();
+        };
+
+        File::create(out.join("memory.x"))
+            .unwrap()
+            .write_all(linker)
+            .unwrap();
+        println!("cargo:rustc-link-search={}", out.display());
+
+        println!("cargo:rerun-if-changed=build.rs");
```
Comment (jonas-schievink) [inline build.rs]: It might be better to always output this line, since otherwise the build script will be rerun on every build I think.

Comment (hannobraun) [reply build.rs]: Good catch, but actually this line is redundant, per https://doc.rust-lang.org/cargo/reference/build-scripts.html. I've pushed another commit that removes it entirely.

Comment (jonas-schievink) [reply build.rs]: Hmm, that says:

> Normally build scripts are re-run if any file inside the crate root changes

And the `cargo:rerun-if-changed=build.rs` note says:

> (unless you want to ignore changes in all other files except for build.rs)

So it's not quite redundant. It probably doesn't hurt to run the build script every time any file changes, but it's also not necessary.

Comment (hannobraun) [reply build.rs]: My interpretation of the `rerun-if-changed` documentation is this:
- By default, the build script is re-run on every change to any file.
- You can restrict this, by issuing `rerun-if-changed`. Then reruns are limited to changes to the specified files.
- **Except** that changes to `build.rs` will always trigger a re-run.
- Therefore `cargo:rerun-if-changed=build.rs` is redundant, unless it is the only `rerun-if-changed` directory issued.

**Comment by arkorobotics** [comment]
Ready to pull?

**Comment by hannobraun** [comment]
Not yet. I think I might have misunderstood what @jonas-schievink was saying. I'll look into it later.

File: build.rs (hunk 1 above)
Comment (hannobraun) [reply build.rs]: Okay, I think I completely misunderstood what you were saying. I was thinking about what happened inside the `if`, and yes, the `cargo:rerun-if-changed=build.rs` is redundant there. It is not redundant if `disable-linker-script` is enabled, as then the linker script is re-run on every file change.

I'm going to push a new commit soon.

**Comment by hannobraun** [comment]
I pushed a new version with the fix @jonas-schievink suggested and squashed the commits. Should be ready to merge now.

**Pull request merged**